src/utils/axiosInstance.ts

```diff
@@ -0,0 +1,10 @@
+import axios from "axios"
+
+const axiosInstance = axios.create({
+  headers: {
+    "Content-Type": "application/json",
+    Authorization: `Bearer ${localStorage.getItem("userToken")}`,
```

Comment:
This will not work if the token was initially null and we signed in. As the value will be null untill we refresh the page. To fix this issue, we need to use interceptors instead. Interceptors run in the middle of a network request, which mean that the last value of the token will be accessed instead of the initiall value of null.